File: php/src/UblRule.php
<?php

declare(strict_types=1);

namespace EPostak;

/**
 * UBL validation rule codes returned in {@see UblValidationException::$rule}.
 *
 * These map to the `error.rule` field in the API's 422 `UBL_VALIDATION_ERROR`
 * response body. Source of truth: `lib/ubl/generate.ts` in the epostak repo.
 *
 * New rules may be added in future API versions — treat these constants as a
 * hint, not a closed enum.
 */
final class UblRule
{
    /** BT-2 — Invoice issue date is mandatory. */
    public const BR_02 = 'BR-02';

    /** BT-27 — Seller name is mandatory. */
    public const BR_05 = 'BR-05';

    /** BT-44 — Buyer name is mandatory. Pass `receiverName` in the request body. */
    public const BR_06 = 'BR-06';

    /** BT-31 / BT-32 — Seller VAT identifier required for VAT-rated invoices. */
    public const BR_11 = 'BR-11';

    /** Invoice must have at least one line. */
    public const BR_16 = 'BR-16';

    /** BT-1 — Invoice number must not be empty. */
    public const BT_1 = 'BT-1';

    /** EndpointID is empty — firm must have DIČ, IČO, or a registered Peppol ID. */
    public const PEPPOL_R008 = 'PEPPOL-R008';
}

File: java/src/main/java/sk/epostak/sdk/UblRule.java
package sk.epostak.sdk;

/**
 * Well-known UBL / Peppol BIS 3.0 schematron rule identifiers that the API
 * returns in a 422 {@code UBL_VALIDATION_ERROR} response.
 *
 * <pre>{@code
 * try {
 *     client.documents().send(req);
 * } catch (UblValidationException e) {
 *     if (UblRule.BR_02.name().equals(e.getRule())) {
 *         System.err.println("Invoice must have a profile identifier");
 *     }
 * }
 * }</pre>
 */
public enum UblRule {

    /**
     * BR-02 — An Invoice shall have a Profile identifier.
     */
    BR_02,

    /**
     * BR-05 — An Invoice shall have an Invoice type code.
     */
    BR_05,

    /**
     * BR-06 — An Invoice shall have a VAT accounting currency code.
     */
    BR_06,

    /**
     * BR-11 — Every Invoice line (BG-25) shall have an Invoice line identifier.
     */
    BR_11,

    /**
     * BR-16 — Each Invoice line (BG-25) shall have an Invoice line net amount.
     */
    BR_16,

    /**
     * BT-1 — The Invoice number (BT-1) shall be a unique identification of the Invoice
     * in the context of the Seller.
     */
    BT_1,

    /**
     * PEPPOL-R008 — The seller's registration identifier scheme identifier must be within
     * the Peppol code list.
     */
    PEPPOL_R008
}
